llvm/include/llvm-c/CAS/PluginAPI_types.h

```diff
@@ -0,0 +1,85 @@
+/*===-- llvm-c/CAS/PluginAPI_Types.h - CAS Plugin Types Interface -*- C -*-===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* The types for the LLVM CAS plugin API.                                     *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_CAS_PLUGINAPI_TYPES_H
+#define LLVM_C_CAS_PLUGINAPI_TYPES_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define LLCASAPI_VERSION_MAJOR 0
+#define LLCASAPI_VERSION_MINOR 1
+
+typedef struct llcasapi_cas_options_s *llcasapi_cas_options_t;
+typedef struct llcasapi_cas_s *llcasapi_cas_t;
+
+/**
+ * Digest hash bytes.
+ */
+typedef struct {
+  const uint8_t *data;
+  size_t size;
+} llcasapi_digest_t;
+
+/**
+ * Data buffer for stored CAS objects.
+ */
+typedef struct {
+  const void *data;
+  size_t size;
+} llcasapi_data_t;
+
+/**
+ * Identifier for a CAS object.
+ */
+typedef struct {
+  uint64_t opaque;
+} llcasapi_objectid_t;
+
+/**
+ * A loaded CAS object.
+ */
+typedef struct {
+  uint64_t opaque;
+} llcasapi_loaded_object_t;
+
+/**
+ * Object references for a CAS object.
+ */
+typedef struct {
+  uint64_t begin;
```

Comment:
What are these integer values representing?  Based on the API functions these values are opaque; maybe they should have "opaque" in the name so no one tries to interpret them manually?

Reply:
Done.